.github/workflows/tf-plan-hash.yml

```diff
@@ -0,0 +1,130 @@
+name: HASH Terraform Plan
+
+# TODO: While this workflow is specific to HASH, we should consider
+#  making this a reusable workflow, parameterized on the directory and environment.
+
+on:
+  workflow_dispatch:
+    inputs:
+      env:
+        type: choice
+        options: ["prod"]
+        description: "Environment to plan"
+        required: true
+        default: "prod"
+
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "infra/terraform/hash/**"
+      - "infra/terraform/modules/**"
+
+env:
+  VAULT_ADDR: ${{ secrets.VAULT_ADDR }}
+
+jobs:
+  matrix-eval:
+    name: Prepare environments to be planned
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - name: Set matrix
+        id: set-matrix
+        env:
+          ENV: ${{ github.event.inputs.env }}
+        run: |
+          # If no env is specified, run both dev and prod
+          if [ "$ENV" == "prod" ]; then
+            echo "matrix=[\"$ENV\"]" >> $GITHUB_OUTPUT
+          else
+            echo "matrix=[\"prod\"]" >> $GITHUB_OUTPUT
```

Comment:
I'm wondering if it's wise to run the production deployment if no environment is specified (regardless of if we also run the dev or not if the variable is missing) – it might be better to force people to be explicit about deploying to prod if that's their intention, to avoid unintended deployments. Or is there some technical reason we need to be able to deploy to prod if the environment variable isn't specified?

Reply:
I guess this comment is better applied to the 'apply' file but you get the point

Reply:
For the `apply` workflow, it wouldn't have any `$ENV` unless it's executed through a `workflow_dispatch`, so we need the default to do merged to main. 

When we dispatch the workflow you have to choose your env, and if we did have a `dev` env we could default to that (we default to `dev` for the internal API apply workflow dispatch)

Reply:
We could maybe read from a [GitHub environment variable](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) to determine if this is a merge to main (e.g. `GITHUB_REF_NAME == "main"`) rather than having it be the default

Reply:
I've addressed this in f21e71353927c3f6af984d7b4111acf9fd9d3db7